packages/cdktf-cli/src/bin/cmds/provider-upgrade.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) HashiCorp, Inc
+// SPDX-License-Identifier: MPL-2.0
+import yargs from "yargs";
+import { requireHandlers } from "./helper/utilities";
+import { Errors } from "../../lib/errors";
+import { BaseCommand } from "./helper/base-command";
+
+class Command extends BaseCommand {
+  public readonly command = "upgrade <provider...>";
+  public readonly describe =
+    "Upgrade one or more Terraform providers in your project.";
+
+  public readonly builder = (args: yargs.Argv) =>
+    args.showHelpOnFail(true).positional("provider", {
+      type: "string",
+      desc: "Name of the provider to upgrade. Can include a version constraint (e.g. aws@~>4.0).",
+      required: true,
+    });
+
+  public async handleCommand(argv: any) {
+    Errors.setScope("provider upgrade");
+    // deferred require to keep cdktf-cli main entrypoint small (e.g. for fast shell completions)
+    const api = requireHandlers();
+    await api.providerUpgrade(argv);
+  }
+}
+
+module.exports = new Command();
```

Comment:
Yes, we do this for code splitting purposes. In order to have a fast code completion the initial js file needs to be small. We put all the handlers into a bigger js file with all the dependencies that is only loaded once a command is found to be running. It's not too much time it takes to load the big js, but enough to be annoying during code completion

Reply:
Hmm, I wonder if this will actually code split though? I don't know much about it, but I thought the only code split that happens is with the [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import). Doesn't matter for this PR though, was just wondering.

Reply:
The dynamic import is done here: https://github.com/hashicorp/terraform-cdk/blob/main/packages/cdktf-cli/src/bin/cmds/helper/utilities.ts#L33